src/content/peerconnection/trickle-ice/js/main.js

```diff
@@ -164,7 +164,33 @@ function iceCallback(event) {
     appendCell(row, c.port);
     appendCell(row, formatPriority(c.priority));
   } else {
-    appendCell(row, 'Done', 7);
+    var result = 'Done';
+    if (servers.length === 1) {
+      var server = JSON.parse(servers[0].value);
+
+      // figure out the types of candidates (host, srflx, relay)
+      var types = pc.localDescription.sdp.split('\r\n').filter(
+        function(line) {
+          return line.indexOf('a=candidate:') === 0;
+        }
+      ).map(function(cand) {
+        return parseCandidate(cand).type;
+      });
+
+      // this trick only works for TURN/UDP
+      if (server.url.indexOf('turn:') === 0 &&
```

Comment:
only that no candidate could be gathered. Since we don't get STUN candidates from a TURN/TCP server it's not possible to decide whether that is due to a unreachable port or  an authentication failure.

Typically I would expect people to test authentication using TURN/UDP and then just reuse those credentials on TURN/TCP.